db/migrate/20181113182615_add_vms_table.rb

```diff
@@ -0,0 +1,26 @@
+class AddVmsTable < ActiveRecord::Migration[5.1]
+  def change
+    create_table :vms do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref
+      t.string :uuid
+      t.string :vendor
+      t.string :name
+      t.string :hostname
+      t.string :description
+      t.string :power_state
+      t.bigint :cpus
```

Comment:
is this enough? I think we also have :virtual_cpus and :sockets?

Reply:
Yeah I figured that is mostly an infra vm thing and we could put in the extra jsonb (which I should add)

Reply:
Okay added a jsonb extra column